staging/src/k8s.io/apiserver/pkg/util/webhook/error.go

```diff
@@ -32,3 +37,32 @@ func (e *ErrCallingWebhook) Error() string {
 	}
 	return fmt.Sprintf("failed calling webhook %q; no further details available", e.WebhookName)
 }
+
+// ErrWebhookRejection represents a webhook properly rejecting a request.
+type ErrWebhookRejection struct {
```

Comment:
this reconstructs the status on every call to Error()/RejectionCode()/ToStatusErr(). If we need a wrapper type, can we just have it contain the pre-constructed status?

Reply:
Error() exists only to satisfy the error interface. I will update the struct to only contain the pre-constructed status